example/poisson/poisson_driver.cpp

```diff
@@ -59,13 +60,84 @@ TaskCollection PoissonDriver::MakeTaskCollection(BlockList_t &blocks) {
   const int num_partitions = pmesh->DefaultNumPartitions();
   TaskRegion &solver_region = tc.AddRegion(num_partitions);
 
+  // setup some reductions
+  // initialize to zero
+  total_mass.val = 0.0;
+  update_norm.val = 0.0;
+  max_rank.val = 0;
+  // we'll also demonstrate how to reduce a vector
+  vec_reduce.val.resize(10);
+  for (int i = 0; i < 10; i++)
+    vec_reduce.val[i] = 0;
+
   for (int i = 0; i < num_partitions; i++) {
     // make/get a mesh_data container for the state
     auto &md = pmesh->mesh_data.GetOrAdd("base", i);
     auto &mdelta = pmesh->mesh_data.GetOrAdd("delta", i);
 
     TaskList &tl = solver_region[i];
 
+    //--- Demo a few reductions
+    // pass a pointer to the variable being reduced into
+    auto loc_red = tl.AddTask(none, poisson_package::SumMass<MeshData<Real>>, md.get(),
+                              &total_mass.val);
+    // make it a regional dependency so dependent tasks can't execute until all lists do
+    // this
+    solver_region.AddRegionalDependencies(0, i, loc_red);
```

Comment:
This example might be made a little clearer by changing the region id's from hard coded numbers to `regiond_id` or `reg_dep_id` and incrementing them as they're used.

Reply:
this is actually a little subtle in the general case, but not here, so I did this.